src/main/java/sirius/web/http/Response.java

```diff
@@ -269,12 +269,28 @@ private void setupCors(DefaultHttpResponse response) {
     }
 
     private void setupCookies(DefaultHttpResponse response) {
-        Collection<Cookie> cookies = wc.getOutCookies();
+        Collection<Cookie> cookies = wc.getOutCookies(isCacheable(response));
         if (cookies != null && !cookies.isEmpty()) {
             response.headers().set(HttpHeaderNames.SET_COOKIE, ServerCookieEncoder.LAX.encode(cookies));
         }
     }
 
+    private boolean isCacheable(DefaultHttpResponse response) {
+        // Check for the obvious - which is our own cache setting...
+        if (cacheSeconds > 0) {
```

Comment:
NPE, cacheSeconds can be null to indicate to use the defaults suggested by the content creator.